pmd-packager/upgrade.sh

```diff
@@ -1,5 +1,7 @@
-VERSION=$(curl -s https://api.github.com/repos/pmd/pmd/releases/latest | jq --raw-output ".tag_name" | sed 's:.*/::')
+VERSION=$(curl -s https://api.github.com/repos/pmd/pmd/releases/latest | grep '.tag_name' | sed 's:.*/::' | sed 's:",::')
 echo $VERSION
+echo -n $VERSION > pmd-packager/pmd-version.txt
+sed -i "" "s|version\>[0-9\.]*\<|version>$VERSION<|g" pmd-packager/pom.xml
```

Comment:
I don't know, whether the change of pom.xml is needed, since you provide the version in the next line via `-Dpmd.version=$VERSION`.

Also, in the next line, you are calling `./mvnw` in the current directory, which implies, you are already in "pmd-packager"....

from where is upgrade.sh called? Currently it's run via `npm run update-pmd`: https://github.com/ChuckJonas/vscode-apex-pmd/blob/95b685cce9eddb690834bb812d70e0b126d95406/package.json#L192

Reply:
Right now, upgrade.sh is just there and in package.json. If this PR is approved, I will submit another PR to do a regularly scheduled github workflow that will execute upgrade.sh, pull the PMD files, unzip them, etc. This is just step 1, and I don't want to bite off more than I can chew at once.

Reply:
As for the change to pom.xml, I cannot comment on that because I was just taking the old code and working with that. I confess that I don't fully understand the existing code, so I was just updating it to have that $VERSION variable, since it will be used when downloading and unzipping the latest PMD version zip.

Reply:
I've run now "npm run update-pmd", here's the output:

```
$ npm run update-pmd

> apex-pmd@0.5.9 update-pmd
> cd pmd-packager && ./upgrade.sh

6.43.0
sed: can't read s|version\>[0-9\.]*\<|version>6.43.0<|g: No such file or directory
sed: can't read pmd-packager/pom.xml: No such file or directory
```

So the sed command obviously doesn't work - and TBH - it's not needed at all, because the version is set in the next line when maven is executed: `./mvnw clean package -Dpmd.dist.bin.baseDirectory=pmd -Dpmd.version=$VERSION`

Actually, looking at the script: It already does what you want to do, you just need to execute it: It already pulls the latest version number and updates pmd under https://github.com/ChuckJonas/vscode-apex-pmd/tree/master/bin/pmd

While looking at the directory, I found another update script: https://github.com/ChuckJonas/vscode-apex-pmd/blob/master/bin/update.sh - not sure, what this is for.

Reply:
I updated the upgrade script.
I didn't create it - I was only updating the existing script. 
The pom.xml file is needed, I believe, so the script updates the version number in that file.
I don't know what the file `bin/update.sh` is about. 
This script just updates the pmd-packager files needed to download PMD to the bin folder. Yes, that update.sh script should probably be moved into upgrade.sh. It appears it's splitting a single action into two scripts. If you're ok with leaving the file in bin for now, we can move those actions into a new PR. Or I can add to this PR and include the unzipping, etc here.

Reply:
After giving it some thought:
I think we need TWO scripts. 

One will update pmd-version.txt to the latest version of PMD. We could also have a setting in the extension to override this, but this pmd-version.txt file would always be the latest version, as we would run it regularly.

The second script will download the specified version of PMD. (Specified is latest, which would become the default extension setting, or a specific version in the extension settings.) For now, we bundle the jar files in the extension.

This way we offload the downloading of the zip file to each project, like .pmdCache. If there's a way to add the bin directory to gitignore automatically, that would help. For now, let's just auto-update the PMD version, and those two scripts can be used in a packaging script.

Open question: How do we unzip only the files that we need?